Formula/libadwaita-1.rb

```diff
@@ -0,0 +1,36 @@
+class Libadwaita1 < Formula
+  desc "Building blocks for modern adaptive GNOME applications"
+  homepage "https://gnome.pages.gitlab.gnome.org/libadwaita/"
+  url "https://download.gnome.org/sources/libadwaita/1.0/libadwaita-1.0.1.tar.xz"
+  sha256 "bb49cf5a09d2e8bc144946c2c3272aecd611667fd027f3808b95d7101ed473d6"
+  license "LGPL-2.1-or-later"
+
+  livecheck do
+    url :stable
+    regex(/libadwaita[._-](1\.([0-8]\d*?)?[02468](?:\.\d+)*?)\.t/i)
+  end
+
+  depends_on "gobject-introspection" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "sassc" => :build
+  depends_on "vala" => :build
+  depends_on "gtk4"
+
+  def install
+    args = std_meson_args + %w[
+      -Dtests=false
+    ]
+
+    mkdir "build" do
+      system "meson", *args, ".."
+      system "ninja", "-v"
+      system "ninja", "install", "-v"
+    end
+  end
+
+  test do
+    system "#{bin}/adwaita-1-demo", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
> If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.

I've done just that and amended the commit. I took the [gtk4 formula's test section](https://github.com/Homebrew/homebrew-core/blob/master/Formula/gtk4.rb#L83) as a template and use a stripped down version of the [upstream's hello-world example program](https://gitlab.gnome.org/GNOME/libadwaita/-/blob/main/examples/hello-world/hello.c). Thanks for your feedback.